<a href="https://colab.research.google.com/github/GPINTOC/Proyecto-2-Coding-Dojo/blob/main/Proyecto2_Parte1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Para esta tarea:**

Creen un cuaderno de Colab donde hayan subido y mostrado el .head() de cada uno de los conjuntos de datos.  Para cada uno de los conjuntos de datos propuestos, respondan las siguientes preguntas:

In [ ]:
import numpy as np
import pandas as pd

In [ ]:
filename1 = '/content/drive/MyDrive/Documentos CodingDojo/parkinson.csv'
df1 = pd.read_csv(filename1)
df1

name  MDVP:Fo(Hz)  MDVP:Fhi(Hz)  MDVP:Flo(Hz)  MDVP:Jitter(%)  \
0    phon_R01_S01_1      119.992       157.302        74.997         0.00784   
1    phon_R01_S01_2      122.400       148.650       113.819         0.00968   
2    phon_R01_S01_3      116.682       131.111       111.555         0.01050   
3    phon_R01_S01_4      116.676       137.871       111.366         0.00997   
4    phon_R01_S01_5      116.014       141.781       110.655         0.01284   
..              ...          ...           ...           ...             ...   
190  phon_R01_S50_2      174.188       230.978        94.261         0.00459   
191  phon_R01_S50_3      209.516       253.017        89.488         0.00564   
192  phon_R01_S50_4      174.688       240.005        74.287         0.01360   
193  phon_R01_S50_5      198.764       396.961        74.904         0.00740   
194  phon_R01_S50_6      214.289       260.277        77.973         0.00567   

     MDVP:Jitter(Abs)  MDVP:RAP  MDVP:PPQ  Jitter:DDP  MDVP:Shimmer  ...  \
0             0.00007   0.00370   0.00554     0.01109       0.04374  ...   
1             0.00008   0.00465   0.00696     0.01394       0.06134  ...   
2             0.00009   0.00544   0.00781     0.01633       0.05233  ...   
3             0.00009   0.00502   0.00698     0.01505       0.05492  ...   
4             0.00011   0.00655   0.00908     0.01966       0.06425  ...   
..                ...       ...       ...         ...           ...  ...   
190           0.00003   0.00263   0.00259     0.00790       0.04087  ...   
191           0.00003   0.00331   0.00292     0.00994       0.02751  ...   
192           0.00008   0.00624   0.00564     0.01873       0.02308  ...   
193           0.00004   0.00370   0.00390     0.01109       0.02296  ...   
194           0.00003   0.00295   0.00317     0.00885       0.01884  ...   

     Shimmer:DDA      NHR     HNR  status      RPDE       DFA   spread1  \
0        0.06545  0.02211  21.033       1  0.414783  0.815285 -4.813031   
1        0.09403  0.01929  19.085       1  0.458359  0.819521 -4.075192   
2        0.08270  0.01309  20.651       1  0.429895  0.825288 -4.443179   
3        0.08771  0.01353  20.644       1  0.434969  0.819235 -4.117501   
4        0.10470  0.01767  19.649       1  0.417356  0.823484 -3.747787   
..           ...      ...     ...     ...       ...       ...       ...   
190      0.07008  0.02764  19.517       0  0.448439  0.657899 -6.538586   
191      0.04812  0.01810  19.147       0  0.431674  0.683244 -6.195325   
192      0.03804  0.10715  17.883       0  0.407567  0.655683 -6.787197   
193      0.03794  0.07223  19.020       0  0.451221  0.643956 -6.744577   
194      0.03078  0.04398  21.209       0  0.462803  0.664357 -5.724056   

      spread2        D2       PPE  
0    0.266482  2.301442  0.284654  
1    0.335590  2.486855  0.368674  
2    0.311173  2.342259  0.332634  
3    0.334147  2.405554  0.368975  
4    0.234513  2.332180  0.410335  
..        ...       ...       ...  
190  0.121952  2.657476  0.133050  
191  0.129303  2.784312  0.168895  
192  0.158453  2.679772  0.131728  
193  0.207454  2.138608  0.123306  
194  0.190667  2.555477  0.148569  

[195 rows x 24 columns]

In [ ]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 195 entries, 0 to 194
Data columns (total 24 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   name              195 non-null    object 
 1   MDVP:Fo(Hz)       195 non-null    float64
 2   MDVP:Fhi(Hz)      195 non-null    float64
 3   MDVP:Flo(Hz)      195 non-null    float64
 4   MDVP:Jitter(%)    195 non-null    float64
 5   MDVP:Jitter(Abs)  195 non-null    float64
 6   MDVP:RAP          195 non-null    float64
 7   MDVP:PPQ          195 non-null    float64
 8   Jitter:DDP        195 non-null    float64
 9   MDVP:Shimmer      195 non-null    float64
 10  MDVP:Shimmer(dB)  195 non-null    float64
 11  Shimmer:APQ3      195 non-null    float64
 12  Shimmer:APQ5      195 non-null    float64
 13  MDVP:APQ          195 non-null    float64
 14  Shimmer:DDA       195 non-null    float64
 15  NHR               195 non-null    float64
 16  HNR               195 non-null    float64
 1

Primera opción: conjunto de datos 1

**1) Fuente de datos**
- https://www.kaggle.com/datasets/debasisdotcom/parkinson-disease-detection

**2) Breve descripción de datos**
- Parkinson es una enfermedad neurodegenerativa que se manifiesta a través de un deterioro del movimiento, incluyendo la presencia de temblores y rigidez. Por lo general, hay un efecto marcado en el habla, que incluye disartria (dificultad para articular sonidos), hipofonía (volumen más bajo) y monotonía (gama de tono reducida). 

**3) ¿Cuál es el objetivo?** 
- El objetivo es contar con un proceso de detección eficaz de Parkinson, que no requiera una visita a la clínica, a través de diferentes categorías de tonos y características de voz

**4) ¿Este es un problema de clasificación o regresión?**
- Clasificación, el status de enfermedad 0-1, saludable o parkinson.

**5) ¿Cuántas características hay?**
- 24 características

**6) ¿Cuántas filas de datos hay?**
- 195 filas 

**7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?**
- A simple vista no detecto valores nulos ni aberrantes, sin embargo veo valores negativos que podrían ocasionar algún tipo de inconveniente

In [7]:
filename2 = '/content/drive/MyDrive/Documentos CodingDojo/cancer de mamas.csv'
df2 = pd.read_csv(filename2)
df2

Age  Gender  Protein1  Protein2  Protein3  Protein4 Tumour_Stage  \
0     42  FEMALE  0.952560   2.15000  0.007972 -0.048340           II   
1     54  FEMALE  0.000000   1.38020 -0.498030 -0.507320           II   
2     63  FEMALE -0.523030   1.76400 -0.370190  0.010815           II   
3     78  FEMALE -0.876180   0.12943 -0.370380  0.132190            I   
4     42  FEMALE  0.226110   1.74910 -0.543970 -0.390210           II   
..   ...     ...       ...       ...       ...       ...          ...   
329   59  FEMALE  0.024598   1.40050  0.024751  0.280320           II   
330   41  FEMALE  0.100120  -0.46547  0.472370 -0.523870            I   
331   54  FEMALE  0.753820   1.64250 -0.332850  0.857860           II   
332   74  FEMALE  0.972510   1.42680 -0.366570 -0.107820           II   
333   66  FEMALE  0.286380   1.39980  0.318830  0.836050           II   

                          Histology ER status PR status HER2 status  \
0     Infiltrating Ductal Carcinoma  Positive  Positive    Negative   
1     Infiltrating Ductal Carcinoma  Positive  Positive    Negative   
2     Infiltrating Ductal Carcinoma  Positive  Positive    Negative   
3     Infiltrating Ductal Carcinoma  Positive  Positive    Negative   
4     Infiltrating Ductal Carcinoma  Positive  Positive    Positive   
..                              ...       ...       ...         ...   
329   Infiltrating Ductal Carcinoma  Positive  Positive    Positive   
330   Infiltrating Ductal Carcinoma  Positive  Positive    Positive   
331   Infiltrating Ductal Carcinoma  Positive  Positive    Negative   
332  Infiltrating Lobular Carcinoma  Positive  Positive    Negative   
333   Infiltrating Ductal Carcinoma  Positive  Positive    Negative   

                    Surgery_type Date_of_Surgery Date_of_Last_Visit  \
0                          Other       20-May-18          26-Aug-18   
1                          Other       26-Apr-18          25-Jan-19   
2                     Lumpectomy       24-Aug-18          08-Apr-20   
3                          Other       16-Nov-18          28-Jul-20   
4                     Lumpectomy       12-Dec-18          05-Jan-19   
..                           ...             ...                ...   
329                   Lumpectomy       15-Jan-19          27-Mar-20   
330  Modified Radical Mastectomy       25-Jul-18          23-Apr-19   
331            Simple Mastectomy       26-Mar-19          11-Oct-19   
332                   Lumpectomy       26-Nov-18          05-Dec-18   
333  Modified Radical Mastectomy       04-Feb-19          10-Aug-19   

    Patient_Status  
0            Alive  
1             Dead  
2            Alive  
3            Alive  
4            Alive  
..             ...  
329          Alive  
330          Alive  
331           Dead  
332          Alive  
333           Dead  

[334 rows x 15 columns]

In [8]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 334 entries, 0 to 333
Data columns (total 15 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Age                 334 non-null    int64  
 1   Gender              334 non-null    object 
 2   Protein1            334 non-null    float64
 3   Protein2            334 non-null    float64
 4   Protein3            334 non-null    float64
 5   Protein4            334 non-null    float64
 6   Tumour_Stage        334 non-null    object 
 7   Histology           334 non-null    object 
 8   ER status           334 non-null    object 
 9   PR status           334 non-null    object 
 10  HER2 status         334 non-null    object 
 11  Surgery_type        334 non-null    object 
 12  Date_of_Surgery     334 non-null    object 
 13  Date_of_Last_Visit  317 non-null    object 
 14  Patient_Status      321 non-null    object 
dtypes: float64(4), int64(1), object(10)
memory usage: 39.3+ K

Segunda opción: conjunto de datos 2

**1) Fuente de datos**

https://www.kaggle.com/datasets/kreeshrajani/breast-cancer-survival-dataset

**2) Breve descripción de datos**

El cáncer de mama es una enfermedad común que se desarrolla por la división rápida de células malignas y provoca que se expanda a otros órganos si no es tratado a tiempo. El dataset contiene información sobre pacientes incluido su ID, edad, sexo y niveles de expresión de cuatro proteínas (Proteína 1, Proteína 2, Proteína 3, Proteína 4). Además incluye el estadio del cáncer de mama del paciente (Tumor_Stage), Histología (tipo de cáncer), estado de ER, PR y HER2, tipo de cirugía, fecha de la cirugía, fecha de la última visita y estado del paciente (vivo/muerto).

**3) ¿Cuál es el objetivo?**

Análisis de la relación entre los niveles de expresión de proteínas, el estadio del cáncer y los resultados de los pacientes. También se puede utilizar para comprender el impacto de los diferentes tipos de cirugías en la supervivencia del paciente y para identificar los factores de riesgo potenciales para la progresión del cáncer de mama.

**4) ¿Este es un problema de clasificación o regresión?**

Clasificación

**5) ¿Cuántas características hay?**

15 columnas

**6) ¿Cuántas filas de datos hay?**

334 filas

**7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?**

Hay datos nulos o desconocidos que habría que imputar, hay fechas descritas como objetos que deberían ser transformadas si es que la columna fuera relevante, habría que realizar one hot encoder a las columnas de clases no ordenadas descritas como objetos y debería definir yo el objetivo del problema, ya que solo es una base de datos con información para ser analizada. 